### PR TITLE
Support showing how a PipelineRun was triggered

### PR DESCRIPTION
**Pull request by andrewballantyne**
Last commit is this functionality: ae6f886a85077a7be2f62f0646b66be91fd8e6ef. Built on the Trigger work done in https://github.com/openshift/console/pull/4984

**Fixes**: 
https://issues.redhat.com/browse/ODC-3323

**Analysis / Root cause**: 
With the addition of Tekton Triggers, we now have a desire to separate _who_ started a Pipeline Run... was it the Trigger or a user?

**Solution Description**: 
Two fold solution:

* For Manual starts - Inject the username into the labels of a Pipeline Run as it is created to carry forward who started it
* For Trigger kicks - Read the provided label from the Operator to link back to the Event Listener

**Screen shots / Gifs for design review**: 

@openshift/team-devconsole-ux

Kapture 2020-04-12 at 13 08 15

**Unit test coverage report**: 

- [ ] Figure out what unit tests, if any we can write here (most functionality is hooks)

**Test setup:**

* Install the Pipeline Operator
* Manually triggering a Pipeline from any button in the UI should tag the username onto the PLR as it is created
* Installing a Trigger and executing it will show the resource link (see me if you need some resources / steps to get up and running fast to test this)

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/src/components/pipelines/modals/start-pipeline/submit-utils.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { merge } from 'lodash';
+import {
+  createPipelineResource,
+  createSecretResource,
+} from '../../pipeline-resource/pipelineResource-utils';
+import { getPipelineRunFromForm } from '../common/utils';
+import { k8sCreate, K8sResourceCommon } from '@console/internal/module/k8s';
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelines/modals/start-pipeline/submit-utils.ts:7]: Please sort import order.

File: frontend/packages/dev-console/src/components/pipelines/modals/triggers/AddTriggerModal.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+type AddTriggerModalProps = ModalComponentProps & {
+  pipeline: Pipeline;
+};
+
+const AddTriggerModal: React.FC<AddTriggerModalProps> = ({ pipeline, close }) => {
+  const initialValues: AddTriggerFormValues = {
+    ...convertPipelineToModalData(pipeline, true),
+    triggerBinding: {
+      name: TRIGGER_BINDING_EMPTY,
+      resource: null,
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelines/modals/triggers/AddTriggerModal.tsx:25]: Why do we have to pass null here?

File: frontend/packages/dev-console/src/components/pipelines/modals/triggers/TriggerBindingSelectorDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+type TriggerBindingSelectorProps = {
+  description?: string;
+  label: string;
+  onChange: (selectedTriggerBinding: TriggerBindingKind) => void;
+  triggerBindingData?: FirehoseResult<TriggerBindingKind[]>;
+};
+
+const TriggerBindingSelectorDropdown: React.FC<TriggerBindingSelectorProps> = (props) => {
+  const { description, label, onChange, triggerBindingData } = props;
+  const triggerBindings = triggerBindingData.data;
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelines/modals/triggers/TriggerBindingSelectorDropdown.tsx:15]: we could do this here and use the collected item down for readability.
```suggestion
  const triggerBindings = triggerBindingData.data.reduce(
        (acc, triggerBinding) => ({
          ...acc,
          [triggerBinding.metadata.name]: triggerBinding.metadata.name,
        }),
        {},
      );
```

File: frontend/packages/dev-console/src/components/pipelines/modals/triggers/TriggerBindingSelectorDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    <DropdownField
+      fullWidth
+      helpText={description}
+      items={triggerBindings.reduce(
+        (acc, triggerBinding) => ({
+          ...acc,
+          [triggerBinding.metadata.name]: triggerBinding.metadata.name,
+        }),
+        {},
+      )}
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelines/modals/triggers/TriggerBindingSelectorDropdown.tsx:27]:
```suggestion
      items={triggerBindings}
```

File: frontend/packages/dev-console/src/components/pipelines/modals/triggers/resource-utils.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+): RouteKind => {
+  const eventListenerName = eventListener.metadata.name;
+  return {
+    apiVersion: apiVersionForModel(RouteModel),
+    kind: RouteModel.kind,
+    metadata: {
+      name: eventListenerName,
+      labels: {
+        'app.kubernetes.io/managed-by': EventListenerModel.kind,
+        'app.kubernetes.io/part-of': 'Triggers',
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelines/modals/triggers/resource-utils.ts:64]: Should the application name will always be `Triggers`?

File: frontend/packages/dev-console/src/components/pipelines/modals/start-pipeline/submit-utils.ts (hunk 1 above)
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelines/modals/start-pipeline/submit-utils.ts:7]: sort order needs to be adjusted.

**Comment by karthikjeeyar** [comment]
Tested it with `developer` user, it works fine. 
image

File: frontend/packages/dev-console/src/components/pipelines/modals/triggers/AddTriggerModal.tsx (hunk 2 above)
Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/pipelines/modals/triggers/AddTriggerModal.tsx:25]: No selected [k8s] resource at form startup -- it's a required field by the types so nothing is hidden.

File: frontend/packages/dev-console/src/components/pipelines/modals/triggers/resource-utils.ts (hunk 5 above)
Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/pipelines/modals/triggers/resource-utils.ts:64]: Oh good call... I can probably improve this. This was a copy of my example YAML that I just slapped variables into -- I'll look at this more critically.

**Comment by karthikjeeyar** [comment]
Looks like some of the tests are failing here.

**Comment by karthikjeeyar** [comment]
Verified both the manual and trigger scenario, works fine.
/lgtm

**Pull request merged**